### ISSUE TITLE
Error  - no app in context  - OCP\Files\NotFoundException:

### ISSUE THREAD

**Issue by wherf** [nextcloud/server#1862]
### Steps to reproduce
1. Running system
   2.
   3.
### Expected behaviour

Tell us what should happen
### Actual behaviour

The log is filled with the error message - Error

no app in context

OCP\Files\NotFoundException:
### Server configuration

**Operating system**:
ubuntu
**Web server:**
apache
**Database:**
mariadb
**PHP version:**

**Nextcloud version:** (see Nextcloud admin page)
10.0.1 (updated per snap)
**Updated from an older Nextcloud/ownCloud or fresh install:**
updated from 10.0
**Where did you install Nextcloud from:**
nextcloud website
**Signing status:**
<details>
<summary>

Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```

</details>

**List of activated apps:**
<details>
<summary>

App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```

</details>

**The content of config/config.php:**
<details>
<summary>

Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

<?php
$CONFIG = array (
  'passwordsalt' =>
  'secret' => 
  'trusted_domains' => 
  array (
    0 => 'localhost',
    1 => '10.7.50.19',
    2 => 'nextcloud',
    3 => 'nextcloud',
    4 => 
  ),
  'datadirectory' => '/var/ncdata',
  'overwrite.cli.url' => 
  'dbtype' => 'mysql',
  'version' => '9.1.0.16',
  'dbname' => 'nextcloud_db',
  'dbhost' => 'localhost',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'oc_ncadmin',
  'dbpassword' => 
  'logtimezone' => 'UTC',
  'installed' => true,
  'instanceid' => 'ocy9r9oj7fyg',
  'appstore.experimental.enabled' => 'true',
  'mail_smtpmode' => 'smtp',
  'mail_smtpauth' => 1,
  'mail_smtpport' => '465',
  'mail_smtphost' => 'smtp.gmail.com',
  'mail_smtpauthtype' => 'LOGIN',
  'mail_from_address' => 
  'mail_domain' => 'gmail.com',
  'mail_smtpsecure' => 'ssl',
  'mail_smtpname' => 
  'mail_smtppassword' => 
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
  'memcache.local' => '\\OC\\Memcache\\Redis',
  'filelocking.enabled' => true,
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'redis' => 
  array (
    'host' => '/var/run/redis/redis.sock',
    'port' => 0,
    'timeout' => 0,
    'dbindex' => 0,
  ),
  'maintenance' => false,
);



Error log shows following

OCP\Files\NotFoundException:
1. /var/www/nextcloud/lib/private/Share20/DefaultShareProvider.php - line 437: OC\Share20\Share->getNode()
2. /var/www/nextcloud/lib/private/Share20/Manager.php - line 878: OC\Share20\DefaultShareProvider->move(Object(OC\Share20\Share), 'Intranet')
3. /var/www/nextcloud/apps/files_sharing/lib/SharedMount.php - line 120: OC\Share20\Manager->moveShare(Object(OC\Share20\Share), 'Intranet')
4. /var/www/nextcloud/apps/files_sharing/lib/SharedMount.php - line 102: OCA\Files_Sharing\SharedMount->updateFileTarget('/Klasse 2010 (2...', Object(OC\Share20\Share))
5. /var/www/nextcloud/apps/files_sharing/lib/SharedMount.php - line 73: OCA\Files_Sharing\SharedMount->verifyMountPoint(Object(OC\Share20\Share), Array)
6. /var/www/nextcloud/apps/files_sharing/lib/MountProvider.php - line 91: OCA\Files_Sharing\SharedMount->__construct('\OC\Files\Stora...', Array, Array, Object(OC\Files\Storage\StorageFactory))
7. /var/www/nextcloud/lib/private/Files/Config/MountProviderCollection.php - line 77: OCA\Files_Sharing\MountProvider->getMountsForUser(Object(OC\User\User), Object(OC\Files\Storage\StorageFactory))
8. [internal function] OC\Files\Config\MountProviderCollection->OC\Files\Config{closure}(Object(OCA\Files_Sharing\MountProvider))
9. /var/www/nextcloud/lib/private/Files/Config/MountProviderCollection.php - line 78: array_map(Object(Closure), Array)
10. /var/www/nextcloud/lib/private/Files/Filesystem.php - line 423: OC\Files\Config\MountProviderCollection->getMountsForUser(Object(OC\User\User))
11. /var/www/nextcloud/apps/files_sharing/lib/sharedstorage.php - line 94: OC\Files\Filesystem initMountPoints('Intranet')
12. /var/www/nextcloud/apps/files_sharing/lib/sharedstorage.php - line 413: OC\Files\Storage\Shared->init()
13. /var/www/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php - line 486: OC\Files\Storage\Shared->getWrapperStorage()
14. /var/www/nextcloud/lib/private/legacy/util.php - line 149: OC\Files\Storage\Wrapper\Wrapper->instanceOfStorage('\OC\Files\Stora...')
15. /var/www/nextcloud/lib/private/Files/Storage/StorageFactory.php - line 101: OC_Util {closure}('/a.thoma/files/...', Object(OC\Files\Storage\Shared), Object(OCA\Files_Sharing\SharedMount))
16. /var/www/nextcloud/lib/private/Files/Mount/MountPoint.php - line 146: OC\Files\Storage\StorageFactory->wrap(Object(OCA\Files_Sharing\SharedMount), Object(OC\Files\Storage\Shared))
17. /var/www/nextcloud/lib/private/Files/Mount/MountPoint.php - line 170: OC\Files\Mount\MountPoint->createStorage()
18. /var/www/nextcloud/lib/private/Files/Utils/Scanner.php - line 121: OC\Files\Mount\MountPoint->getStorage()
19. /var/www/nextcloud/apps/files/lib/BackgroundJob/ScanFiles.php - line 88: OC\Files\Utils\Scanner->backgroundScan('')
20. /var/www/nextcloud/apps/files/lib/BackgroundJob/ScanFiles.php - line 112: OCA\Files\BackgroundJob\ScanFiles->runScanner(Object(OC\User\User))
21. /var/www/nextcloud/lib/private/BackgroundJob/Job.php - line 53: OCA\Files\BackgroundJob\ScanFiles->run(NULL)
22. /var/www/nextcloud/lib/private/BackgroundJob/TimedJob.php - line 54: OC\BackgroundJob\Job->execute(Object(OC\BackgroundJob\JobList), Object(OC\Log))
23. /var/www/nextcloud/cron.php - line 121: OC\BackgroundJob\TimedJob->execute(Object(OC\BackgroundJob\JobList), Object(OC\Log))
24. {main}


Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
#### LDAP configuration (delete this part if not used)

<details>
<summary>

LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

</details>
### Client configuration

**Browser:**

**Operating system:**
#### Web server error log

<details>
<summary>

Web server error log</summary>

```
Insert your webserver log here
```

</details>
#### Nextcloud log (data/nextcloud.log)

<details>
<summary>

Nextcloud log</summary>

```
Insert your Nextcloud log here
```

</details>
#### Browser log

<details>
<summary>

Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```

</details>

**Comment by wherf** [comment]
error log shows following

OCP\Files\NotFoundException:
1. /var/www/nextcloud/lib/private/Share20/DefaultShareProvider.php - line 437: OC\Share20\Share->getNode()
2. /var/www/nextcloud/lib/private/Share20/Manager.php - line 878: OC\Share20\DefaultShareProvider->move(Object(OC\Share20\Share), 'Intranet')
3. /var/www/nextcloud/apps/files_sharing/lib/SharedMount.php - line 120: OC\Share20\Manager->moveShare(Object(OC\Share20\Share), 'Intranet')
4. /var/www/nextcloud/apps/files_sharing/lib/SharedMount.php - line 102: OCA\Files_Sharing\SharedMount->updateFileTarget('/Klasse 2010 (2...', Object(OC\Share20\Share))
5. /var/www/nextcloud/apps/files_sharing/lib/SharedMount.php - line 73: OCA\Files_Sharing\SharedMount->verifyMountPoint(Object(OC\Share20\Share), Array)
6. /var/www/nextcloud/apps/files_sharing/lib/MountProvider.php - line 91: OCA\Files_Sharing\SharedMount->__construct('\OC\Files\Stora...', Array, Array, Object(OC\Files\Storage\StorageFactory))
7. /var/www/nextcloud/lib/private/Files/Config/MountProviderCollection.php - line 77: OCA\Files_Sharing\MountProvider->getMountsForUser(Object(OC\User\User), Object(OC\Files\Storage\StorageFactory))
8. [internal function] OC\Files\Config\MountProviderCollection->OC\Files\Config{closure}(Object(OCA\Files_Sharing\MountProvider))
9. /var/www/nextcloud/lib/private/Files/Config/MountProviderCollection.php - line 78: array_map(Object(Closure), Array)
10. /var/www/nextcloud/lib/private/Files/Filesystem.php - line 423: OC\Files\Config\MountProviderCollection->getMountsForUser(Object(OC\User\User))
11. /var/www/nextcloud/apps/files_sharing/lib/sharedstorage.php - line 94: OC\Files\Filesystem initMountPoints('Intranet')
12. /var/www/nextcloud/apps/files_sharing/lib/sharedstorage.php - line 413: OC\Files\Storage\Shared->init()
13. /var/www/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php - line 486: OC\Files\Storage\Shared->getWrapperStorage()
14. /var/www/nextcloud/lib/private/legacy/util.php - line 149: OC\Files\Storage\Wrapper\Wrapper->instanceOfStorage('\OC\Files\Stora...')
15. /var/www/nextcloud/lib/private/Files/Storage/StorageFactory.php - line 101: OC_Util {closure}('/a.thoma/files/...', Object(OC\Files\Storage\Shared), Object(OCA\Files_Sharing\SharedMount))
16. /var/www/nextcloud/lib/private/Files/Mount/MountPoint.php - line 146: OC\Files\Storage\StorageFactory->wrap(Object(OCA\Files_Sharing\SharedMount), Object(OC\Files\Storage\Shared))
17. /var/www/nextcloud/lib/private/Files/Mount/MountPoint.php - line 170: OC\Files\Mount\MountPoint->createStorage()
18. /var/www/nextcloud/lib/private/Files/Utils/Scanner.php - line 121: OC\Files\Mount\MountPoint->getStorage()
19. /var/www/nextcloud/apps/files/lib/BackgroundJob/ScanFiles.php - line 88: OC\Files\Utils\Scanner->backgroundScan('')
20. /var/www/nextcloud/apps/files/lib/BackgroundJob/ScanFiles.php - line 112: OCA\Files\BackgroundJob\ScanFiles->runScanner(Object(OC\User\User))
21. /var/www/nextcloud/lib/private/BackgroundJob/Job.php - line 53: OCA\Files\BackgroundJob\ScanFiles->run(NULL)
22. /var/www/nextcloud/lib/private/BackgroundJob/TimedJob.php - line 54: OC\BackgroundJob\Job->execute(Object(OC\BackgroundJob\JobList), Object(OC\Log))
23. /var/www/nextcloud/cron.php - line 121: OC\BackgroundJob\TimedJob->execute(Object(OC\BackgroundJob\JobList), Object(OC\Log))
24. {main}

**Comment by nickvergessen** [comment]
@rullzer

**Comment by enboig** [comment]
Same issue with nextcloud 10.0.1 upgraded from owncloud 9.0

Any hint how to debug/solve the problem?

**Comment by wherf** [comment]
Nope, I have not had any feedback and am still waiting for a response. My system is running but the log file is getting bigger and bigger.

At some point we need a resolution

Von: enboig [mailto:notifications@github.com]
Gesendet: Freitag, 11. November 2016 08:59
An: nextcloud/server
Cc: Wolf Dieter Herf; Author
Betreff: Re: [nextcloud/server] Error - no app in context - OCP\Files\NotFoundException: (#1862)


Same issue with nextcloud 10.0.1 upgraded from owncloud 9.0

Any hint how to debug/solve the problem?

**Comment by rullzer** [comment]
As far as I can tell this is fixed in NC11.
I'll try to see if I can find a quick fix for NC10

**Comment by MorrisJobke** [comment]
Fixed wiht #2091

**Comment by ccoenen** [comment]
I'm seeing something very similar after the upgrade to 12.0.3

**Comment by IPlayZed** [comment]
This is not fixed, running 24.0.2:

```
[no app in context] Error: OCP\Files\NotFoundException:  at <<closure>>

 0. /srv/nextcloud/lib/private/Files/Node/Node.php line 344
    OC\Files\Node\Node->getFileInfo()
 1. /srv/nextcloud/lib/private/Files/Node/Node.php line 158
    OC\Files\Node\Node->getMountPoint()
 2. /srv/nextcloud/apps/maps/lib/Hooks/FileHooks.php line 182
    OC\Files\Node\Node->getStorage()
 3. /srv/nextcloud/apps/maps/lib/Hooks/FileHooks.php line 90
    OCA\Maps\Hooks\FileHooks->isUserNode()
 4. <<closure>>
    OCA\Maps\Hooks\FileHooks->OCA\Maps\Hooks\{closure}("*** sensitive parameters replaced ***")
 5. /srv/nextcloud/lib/private/Hooks/EmitterTrait.php line 106
    call_user_func_array()
 6. /srv/nextcloud/lib/private/Hooks/PublicEmitter.php line 40
    OC\Hooks\BasicEmitter->emit()
 7. /srv/nextcloud/lib/private/Files/Node/Root.php line 143
    OC\Hooks\PublicEmitter->emit()
 8. <<closure>>
    OC\Files\Node\Root->emit()
 9. /srv/nextcloud/lib/private/Files/Node/LazyFolder.php line 72
    call_user_func_array()
10. /srv/nextcloud/lib/private/Files/Node/LazyFolder.php line 100
    OC\Files\Node\LazyFolder->__call()
11. /srv/nextcloud/lib/private/Files/Node/HookConnector.php line 190
    OC\Files\Node\LazyFolder->emit()
12. /srv/nextcloud/lib/private/legacy/OC_Hook.php line 106
    OC\Files\Node\HookConnector->postRename()
13. /srv/nextcloud/lib/private/Files/View.php line 866
    OC_Hook::emit()
14. /srv/nextcloud/apps/dav/lib/Connector/Sabre/Directory.php line 455
    OC\Files\View->rename()
15. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/Tree.php line 160
    OCA\DAV\Connector\Sabre\Directory->moveInto()
16. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 612
    Sabre\DAV\Tree->move()
17. /srv/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpMove()
18. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit()
19. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod()
20. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
21. /srv/nextcloud/apps/dav/lib/Server.php line 352
    Sabre\DAV\Server->exec()
22. /srv/nextcloud/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
23. /srv/nextcloud/remote.php line 166
    require_once("/srv/nextcloud/ ... p")

MOVE /nextcloud/remote.php/dav/files/admin/InstantUpload/Horny/SpankBang.com_deep%2Banal%2Bdildo%2Bgaping_1080p.mp4
from 188.157.75.68 by admin at 2022-07-08T21:22:55+00:00
```

**Comment by IPlayZed** [comment]
FYI, this is in the error log when using WebDAV:
```
[webdav] Fatal: OCP\Lock\LockedException: "InstantUpload/<private-filename>.mp4" is locked at <<closure>>

0. /srv/nextcloud/lib/private/Files/View.php line 808
   OC\Files\View->changeLock()
1. /srv/nextcloud/apps/dav/lib/Connector/Sabre/Node.php line 161
   OC\Files\View->rename()
2. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/Tree.php line 153
   OCA\DAV\Connector\Sabre\Node->setName()
3. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 612
   Sabre\DAV\Tree->move()
4. /srv/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
   Sabre\DAV\CorePlugin->httpMove()
5. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 472
   Sabre\DAV\Server->emit()
6. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 253
   Sabre\DAV\Server->invokeMethod()
7. /srv/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 321
   Sabre\DAV\Server->start()
8. /srv/nextcloud/apps/dav/appinfo/v1/webdav.php line 83
   Sabre\DAV\Server->exec()
9. /srv/nextcloud/remote.php line 166
   require_once("/srv/nextcloud/ ... p")

MOVE /nextcloud/remote.php/webdav/InstantUpload/<private-filename>.mp4
from <my-ip> by admin at 2022-07-09T00:00:57+00:00
```
The **<private>** parts are the filename and the IP, ofc I do not include that in the logs for privacy.
